### PR TITLE
Fix metric/spans count, add tests for nil entries in the slices

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes in this PR:
* Handle correctly nil Resource[Spans|Metrics], InstrumentationLibrary[Spans|Metrics], Span|Metric, MetricDataPoints.
* Found a bug that the OTLP MetricDescriptor.Labels were ignored in Internal->OC
* Rename translator/internaldata/{oc_testdata.go => oc_testdata_test.go} to not count these lines in coverage.

Work left for slices:
* Tests and fixes for nil Events, Links, Buckets, ValueAtPercentile

Work left for maps:
* Define what should be the behavior if a nil entry in the "slice map representation".
* Add tests for these cases.

### CODE REVIEW

File: translator/internaldata/metrics_to_oc.go [hunk 1]
```diff
@@ -100,21 +112,41 @@ func collectLabelKeys(metric data.Metric) *labelKeys {
 
 	// First, collect a set of all labels present in the metric
 	keySet := make(map[string]struct{})
+	md := metric.MetricDescriptor()
+	if !md.IsNil() {
+		addLabelKeys(keySet, md.LabelsMap())
```
Comment (tigrannajaryan) [inline translator/internaldata/metrics_to_oc.go:117]: So this corresponds to the comment for this field that says it should apply to all points: https://github.com/open-telemetry/opentelemetry-proto/blob/e6c3c4a74d57f870a0d781bada02cb2b2c497d14/opentelemetry/proto/metrics/v1/metrics.proto#L177

File: translator/internaldata/metrics_to_oc.go [hunk 2]
```diff
@@ -398,6 +444,23 @@ func labelValuesToOC(labels data.StringMap, labelKeys *labelKeys) []*ocmetrics.L
 		labelValue.HasValue = true
 	}
 
+	// Visit all defined labels in the MetricDescriptor and override defaults with actual values
```
Comment (tigrannajaryan) [inline translator/internaldata/metrics_to_oc.go:447]: Shouldn't we do this before previous section? I assume if the same label is defined both in `MetricDescriptor` and in the particular data point then the label in data point should have the effect?

File: translator/internaldata/oc_to_metrics.go [hunk 3]
```diff
@@ -236,6 +242,18 @@ func setDataPoints(ocMetric *ocmetrics.Metric, metric data.Metric) {
...
+	}
+	if metric.DoubleDataPoints().Len() != 0 {
+		metric.DoubleDataPoints().Resize(doubleDataPointsNum)
+	}
+	if metric.HistogramDataPoints().Len() != 0 {
+		metric.HistogramDataPoints().Resize(histogramDataPointsNum)
+	}
+	if metric.SummaryDataPoints().Len() != 0 {
+		metric.SummaryDataPoints().Resize(summaryDataPointsNum)
+	}
```
Comment (tigrannajaryan) [inline translator/internaldata/oc_to_metrics.go:256]: I don't understand why we have this code. Is it to remove nil data points?

**Comment by tigrannajaryan** [comment]
@bogdandrutu Please use the PR description that you create as the commit message as well. I typically copy/paste them when merging, but someone may forget when merging. We want nice commit messages.
